Discussion/Discussion/Presentation/DiscussionTopics/DiscussionTopicsViewModel.swift

```diff
@@ -31,46 +32,54 @@ public class DiscussionTopicsViewModel: ObservableObject {
     let router: DiscussionRouter
     let analytics: DiscussionAnalytics
     let config: ConfigProtocol
-    
-    public init(title: String,
-                interactor: DiscussionInteractorProtocol,
-                router: DiscussionRouter,
-                analytics: DiscussionAnalytics,
-                config: ConfigProtocol) {
+
+    public init(
+        title: String,
+        interactor: DiscussionInteractorProtocol,
+        router: DiscussionRouter,
+        analytics: DiscussionAnalytics,
+        config: ConfigProtocol
+    ) {
         self.title = title
         self.interactor = interactor
         self.router = router
         self.analytics = analytics
         self.config = config
     }
-    
+
     func generateTopics(topics: Topics?) -> [DiscussionTopic] {
         var result = [
             DiscussionTopic(
                 name: DiscussionLocalization.Topics.allPosts,
                 action: {
-                    self.analytics.discussionAllPostsClicked(courseId: self.courseID,
-                                                                    courseName: self.title)
+                    self.analytics.discussionAllPostsClicked(
```

Comment:
As we are here how about using weak self in block?

Reply:
We don't need weak self for button action from view